src/main/webapp/js/ideas/ideaAddNew.controller.js

```diff
@@ -35,30 +52,56 @@
 				);
 			}//for
 
-		    var request = {
-		        id:null,
-		        description:data.description,
-		        title:data.title,
-		        createdAt:new Date().getTime(),
-		        lastModifiedAt:new Date().getTime(),
-		        tags: tags,
-		        latitude: (ideaCoords)?ideaCoords[1]:0,
-		        longitude: (ideaCoords)?ideaCoords[0]:0
-		    };
-
-            ideasFactory.insertIdea(request).then(
-			   //success
-			   function( value )
-			   {
-				$window.location.href = '#home';
-			   },
-			   //error
-			   function( error ){
-				alert("Ошибка создания идеии: " + error.statusText);
-			   }
-			 );
+			if($scope.imageExist == true){
+				//load file to imgur
+				imgur.setAPIKey('Client-ID c62cfae02efe4c0');
+				imgur.upload($scope.imageFile).then(function then(model) {
+						console.log('Your adorable cat be here: ' + model.link);
+
+						var request = {
```

Comment:
May be part "var request =... $scope.insertIdea..." can be moved out from if{}else{} statements to do not duplicate it?